src/include/planner/logical_plan/logical_operator/base_logical_extend.h

```diff
@@ -19,6 +19,7 @@ class BaseLogicalExtend : public LogicalOperator {
     inline std::shared_ptr<binder::NodeExpression> getNbrNode() const { return nbrNode; }
     inline std::shared_ptr<binder::RelExpression> getRel() const { return rel; }
     inline common::RelDirection getDirection() const { return direction; }
+    inline bool isDirected() const { return rel->isDirected(); }
```

Comment:
I think we shouldn't use this function to check whether extend is UNDIRECTED or not. Here's my high level understanding.

At grammar level, we use ArrowDirection to indicate direction.

At binding level, we interpret ArrowDirection to resolve src and dst nodes. If ArrowDirection is UNDIRECTED, we introduce a new field in RelExpression to keep that information. E.g.
```
(a)->(b) src = a, dst = b, directed = true
(a)<-(b) src = b, dst = a, directed = true
(a) - (b) src = a, dst = b, directed = false
```

So far the design makes sense.

At planning level, a logical extend need 3 information to figure out how to scan rel table. That is, `boundNode`, `nbrNode` and `RelDirection`. 

E.g. Given query `(a)->(b) src = a, dst = b, directed = true`
We can have different extends to compute this query
```
Plan: Scan(a)-> Extend(a->b)
boundNode = a, nbrNode = b, RelDirection = FWD
```
or 
```
Plan: Scan(b) -> Extend(b<-a)
boundNode = b, nbrNode = a, RelDirection = BWD
```

Now consider undirected query `(a) - (b) src = a, dst = b, directed = false`
I want to enumerate similar plans as above
```
Plan: Scan(a)-> Extend(a->b)
boundNode = a, nbrNode = b, RelDirection = UNDIRECTED
```
or 
```
Plan: Scan(b) -> Extend(b<-a)
boundNode = b, nbrNode = a, RelDirection = UNDIRECTED
```
(P.S.) I think calling it BOTH is better than UNDIRECTED because the semantic is the union of both FWD and BWD.

In order to do this, we should first differentiate the RelDirection being used in planning and storage. 

In planning, we should have a ExtendDirection = {FWD, BWD, BOTH} that is used in logicalExtend.

In storage, we should rename RelDirection to RelDataDirection and it contains {FWD, BWD}.

Finally, in mapLogicalExtendToPhysical you want to use ExtendDirection to fetch a RelData in the same direction, 

i.e. ExtendDirection::FWD -> fetch adjIndex correspond to RelDataDirection::FWD
ExtendDirection::BWD -> fetch adjIndex correspond to RelDataDirection::BWD
ExtendDirection::BOTH -> fetch adjIndex correspond to RelDataDirection::FWD and RelDataDirection::BWD

Reply:
Done, except for mapLogicalExtendToPhysical. Currently just left it to use RelDataDirection, I think this change is better suited for next PR.